Flaky Makefile for coq-bignums

<!-- Thank you for reporting a bug to Coq! -->

#### Description of the problem
The Makefile generated by `coq_makefile` sometimes fails for the package `coq-bignums`. Note that I have not reproduced on Coq 8.8 or 8.9, or with OCaml 4.07.

<!-- If you can, it's helpful to provide self-contained example of some code
that reproduces the bug. If not, a link to a larger example is also helpful. -->

#### Coq Version
* with Coq 8.6: `opam list` and `opam install`:
```
# Packages matching: installed
# Name              # Installed       # Synopsis
base-bigarray       base
base-num            base              Num library distributed with the OCaml compiler
base-threads        base
base-unix           base
camlp5              7.06.10-g84ce6cc4 Preprocessor-pretty-printer of OCaml
conf-m4             1                 Virtual package relying on m4
coq                 8.6               Formal proof management system.
num                 0                 The Num library for arbitrary-precision integer and rational arithmetic
ocaml               4.05.0            The OCaml compiler (virtual package)
ocaml-base-compiler 4.05.0            Official 4.05.0 release
ocaml-config        1                 OCaml Switch Configuration
ocamlfind           1.8.1             A library manager for OCaml
The following actions will be performed:
  - install coq-bignums 8.6.0
<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
Processing  1/1: [coq-bignums.8.6.0: http]
[coq-bignums.8.6.0] downloaded from https://github.com/coq/bignums/archive/v8.6.0.tar.gz
Processing  1/1:
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
Processing  1/2: [coq-bignums: make]
+ /home/bench/.opam/opam-init/hooks/sandbox.sh "build" "make" "-j7" (CWD=/home/bench/.opam/ocaml-base-compiler.4.05.0/.opam-switch/build/coq-bignums.8.6.0)
- CAMLDEP bignums_syntax.ml
- COQDEP bignums_syntax_plugin.mlpack
- COQDEP BigZ/BigZ.v
- COQDEP BigZ/ZMake.v
- COQDEP BigNumPrelude.v
- COQDEP CyclicDouble/DoubleCyclic.v
- COQDEP CyclicDouble/DoubleLift.v
- COQDEP CyclicDouble/DoubleSqrt.v
- COQDEP CyclicDouble/DoubleDivn1.v
- COQDEP CyclicDouble/DoubleSub.v
- COQDEP CyclicDouble/DoubleMul.v
- COQDEP CyclicDouble/DoubleDiv.v
- COQDEP CyclicDouble/DoubleBase.v
- COQDEP CyclicDouble/DoubleAdd.v
- COQDEP BigQ/QMake.v
- COQDEP BigQ/BigQ.v
- COQDEP SpecViaZ/ZSigZAxioms.v
- COQDEP SpecViaZ/NSigNAxioms.v
- COQDEP SpecViaZ/ZSig.v
- COQDEP SpecViaQ/QSig.v
- COQDEP SpecViaZ/NSig.v
- ocaml BigN/NMake_gen.ml > BigN/NMake_gen.v || (RV=$?; rm -f BigN/NMake_gen.v; exit ${RV})
- COQDEP BigN/BigN.v
- COQDEP BigN/Nbasic.v
- COQDEP BigN/NMake.v
- COQDEP BigN/NMake_gen.v
- CAMLC -c bignums_syntax.ml
- CAMLOPT -c -for-pack Bignums_syntax_plugin bignums_syntax.ml
- COQC SpecViaZ/NSig.v
- COQC SpecViaZ/ZSig.v
- COQC SpecViaQ/QSig.v
- CAMLC -pack -o bignums_syntax_plugin.cmo
- File "_none_", line 1:
- Error: Corrupted compiled interface
- bignums_syntax.cmi
- Makefile:435: recipe for target 'bignums_syntax_plugin.cmo' failed
- make: *** [bignums_syntax_plugin.cmo] Error 2
- make: *** Waiting for unfinished jobs....
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Bigint, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module CErrors, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Coqlib, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Globnames, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Libnames, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Loc, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Mltop, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Names, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Nametab, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Notation, and its interface was not compiled with -opaque
- File "_none_", line 1:
- Warning 58: no cmx file was found in path for module Pp, and its interface was not compiled with -opaque
[ERROR] The compilation of coq-bignums failed at "/home/bench/.opam/opam-init/hooks/sandbox.sh build make -j7".
#=== ERROR while compiling coq-bignums.8.6.0 ==================================#
# context              2.0.1 | linux/x86_64 | ocaml-base-compiler.4.05.0 | file:///home/bench/run/opam-coq-archive/released
# path                 ~/.opam/ocaml-base-compiler.4.05.0/.opam-switch/build/coq-bignums.8.6.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build make -j7
# exit-code            2
# env-file             ~/.opam/log/coq-bignums-372-d0f245.env
# output-file          ~/.opam/log/coq-bignums-372-d0f245.out
### output ###
# [...]
# File "_none_", line 1:
# Warning 58: no cmx file was found in path for module Loc, and its interface was not compiled with -opaque
# File "_none_", line 1:
# Warning 58: no cmx file was found in path for module Mltop, and its interface was not compiled with -opaque
# File "_none_", line 1:
# Warning 58: no cmx file was found in path for module Names, and its interface was not compiled with -opaque
# File "_none_", line 1:
# Warning 58: no cmx file was found in path for module Nametab, and its interface was not compiled with -opaque
# File "_none_", line 1:
# Warning 58: no cmx file was found in path for module Notation, and its interface was not compiled with -opaque
# File "_none_", line 1:
# Warning 58: no cmx file was found in path for module Pp, and its interface was not compiled with -opaque
<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build coq-bignums 8.6.0
+- 
- No changes have been performed
```
* with Coq 8.7: `opam list` and `opam install`:
```
# Packages matching: installed
# Name              # Installed       # Synopsis
base-bigarray       base
base-num            base              Num library distributed with the OCaml compiler
base-ocamlbuild     base              OCamlbuild binary and libraries distributed with the OCaml compiler
base-threads        base
base-unix           base
camlp5              7.06.10-g84ce6cc4 Preprocessor-pretty-printer of OCaml
conf-m4             1                 Virtual package relying on m4
coq                 8.7.0             Formal proof management system.
num                 0                 The Num library for arbitrary-precision integer and rational arithmetic
ocaml               4.02.3            The OCaml compiler (virtual package)
ocaml-base-compiler 4.02.3            Official 4.02.3 release
ocaml-config        1                 OCaml Switch Configuration
ocamlfind           1.8.1             A library manager for OCaml
The following actions will be performed:
  - install coq-bignums 8.7.0
<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
Processing  1/1: [coq-bignums.8.7.0: http]
[coq-bignums.8.7.0] downloaded from https://github.com/coq/bignums/archive/V8.7.0.tar.gz
Processing  1/1:
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
Processing  1/2: [coq-bignums: make]
+ /home/bench/.opam/opam-init/hooks/sandbox.sh "build" "make" "-j7" (CWD=/home/bench/.opam/ocaml-base-compiler.4.02.3/.opam-switch/build/coq-bignums.8.7.0)
- make: Circular Make <- Makefile.coq dependency dropped.
- coq_makefile -f Make -o Makefile.coq
- Warning: -extra and -extra-phony are deprecated.
- Warning: Write the extra targets in Makefile.coq.local.
- 
- make -f Makefile.coq Makefile
- make[1]: Entering directory '/home/bench/.opam/ocaml-base-compiler.4.02.3/.opam-switch/build/coq-bignums.8.7.0'
- COQDEP bignums_syntax_plugin.mlpack
- CAMLDEP bignums_syntax.ml
- COQDEP BigZ/BigZ.v
- COQDEP CyclicDouble/DoubleLift.v
- COQDEP BigZ/ZMake.v
- COQDEP BigNumPrelude.v
- COQDEP CyclicDouble/DoubleCyclic.v
- COQDEP CyclicDouble/DoubleSqrt.v
- COQDEP CyclicDouble/DoubleDivn1.v
- COQDEP CyclicDouble/DoubleSub.v
- COQDEP CyclicDouble/DoubleMul.v
- COQDEP CyclicDouble/DoubleAdd.v
- COQDEP CyclicDouble/DoubleDiv.v
- COQDEP CyclicDouble/DoubleBase.v
- COQDEP BigQ/BigQ.v
- COQDEP BigQ/QMake.v
- COQDEP SpecViaQ/QSig.v
- COQDEP SpecViaZ/NSigNAxioms.v
- COQDEP SpecViaZ/ZSigZAxioms.v
- COQDEP BigN/BigN.v
- COQDEP SpecViaZ/ZSig.v
- COQDEP SpecViaZ/NSig.v
- ocaml BigN/NMake_gen.ml > BigN/NMake_gen.v || (RV=$?; rm -f BigN/NMake_gen.v; exit ${RV})
- COQDEP BigN/NMake.v
- COQDEP BigN/Nbasic.v
- *** Warning: in file BigN/NMake.v, library NMake_gen is required and has not been found in the loadpath!
- *** Warning: in file BigN/NMake.v, library NMake_gen is required and has not been found in the loadpath!
- COQDEP BigN/NMake_gen.v
- make[1]: Nothing to be done for 'Makefile'.
- make[1]: Leaving directory '/home/bench/.opam/ocaml-base-compiler.4.02.3/.opam-switch/build/coq-bignums.8.7.0'
- make -f Makefile.coq all
- make[1]: Entering directory '/home/bench/.opam/ocaml-base-compiler.4.02.3/.opam-switch/build/coq-bignums.8.7.0'
- CAMLOPT -c -for-pack Bignums_syntax_plugin bignums_syntax.ml
- COQC SpecViaZ/NSig.v
- COQC SpecViaQ/QSig.v
- COQC SpecViaZ/ZSig.v
- CAMLOPT -pack -o bignums_syntax_plugin.cmx
- CAMLOPT -a -o bignums_syntax_plugin.cmxa
- CAMLOPT -shared -o bignums_syntax_plugin.cmxs
- COQC SpecViaZ/NSigNAxioms.v
- COQC SpecViaZ/ZSigZAxioms.v
- COQC BigNumPrelude.v
- COQC BigZ/ZMake.v
- COQC CyclicDouble/DoubleBase.v
- COQC BigQ/QMake.v
- COQC CyclicDouble/DoubleMul.v
- COQC CyclicDouble/DoubleLift.v
- COQC CyclicDouble/DoubleSqrt.v
- COQC CyclicDouble/DoubleAdd.v
- COQC CyclicDouble/DoubleSub.v
- COQC CyclicDouble/DoubleDivn1.v
- COQC CyclicDouble/DoubleDiv.v
- COQC CyclicDouble/DoubleCyclic.v
- COQC BigN/Nbasic.v
- COQC BigN/NMake.v
- COQC BigN/NMake_gen.v
- File "./BigN/NMake.v", line 20, characters 32-41:
- Error: Unable to locate library NMake_gen.
- 
- Makefile.coq:639: recipe for target 'BigN/NMake.vo' failed
- make[2]: *** [BigN/NMake.vo] Error 1
- make[2]: *** Waiting for unfinished jobs....
- Makefile.coq:312: recipe for target 'all' failed
- make[1]: *** [all] Error 2
- make[1]: Leaving directory '/home/bench/.opam/ocaml-base-compiler.4.02.3/.opam-switch/build/coq-bignums.8.7.0'
- Makefile:2: recipe for target 'all' failed
- make: *** [all] Error 2
[ERROR] The compilation of coq-bignums failed at "/home/bench/.opam/opam-init/hooks/sandbox.sh build make -j7".
#=== ERROR while compiling coq-bignums.8.7.0 ==================================#
# context              2.0.1 | linux/x86_64 | ocaml-base-compiler.4.02.3 | file:///home/bench/run/opam-coq-archive/released
# path                 ~/.opam/ocaml-base-compiler.4.02.3/.opam-switch/build/coq-bignums.8.7.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build make -j7
# exit-code            2
# env-file             ~/.opam/log/coq-bignums-8305-d0f245.env
# output-file          ~/.opam/log/coq-bignums-8305-d0f245.out
### output ###
# [...]
# COQC BigN/NMake_gen.v
# File "./BigN/NMake.v", line 20, characters 32-41:
# Error: Unable to locate library NMake_gen.
# 
# Makefile.coq:639: recipe for target 'BigN/NMake.vo' failed
# make[2]: *** [BigN/NMake.vo] Error 1
# make[2]: *** Waiting for unfinished jobs....
# Makefile.coq:312: recipe for target 'all' failed
# make[1]: *** [all] Error 2
# make[1]: Leaving directory '/home/bench/.opam/ocaml-base-compiler.4.02.3/.opam-switch/build/coq-bignums.8.7.0'
# Makefile:2: recipe for target 'all' failed
# make: *** [all] Error 2
<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build coq-bignums 8.7.0
+- 
- No changes have been performed
```
* I have not seen this error with Coq 8.8 or 8.9, but this is hard to reproduce.
<!-- You can get this information by running `coqtop -v`. If relevant, please
also include your operating system. -->

--- a/doc/sphinx/practical-tools/utilities.rst
+++ b/doc/sphinx/practical-tools/utilities.rst
@@ -515,6 +515,11 @@ To build, say, two targets foo.vo and bar.vo in parallel one can use
   + ``-extra-phony`` and ``-extra`` are deprecated. To provide additional target
     (``.PHONY`` or not) please use ``CoqMakefile.local``.
 
+.. note::
+
+  Due to limitations with the compilation chain, makefiles generated
+  by ``coq_makefile`` won't correctly compile OCaml plugins with OCaml
+  < 4.07.0 when using more than one job (``-j N`` for ``N > 1``).
 
 Precompiling for ``native_compute``
 +++++++++++++++++++++++++++++++++++

